packages/mdc-menu/index.js

```diff
@@ -111,85 +138,106 @@ class MDCMenu extends MDCComponent {
     }
   }
 
-  /** @param {number} index */
-  set selectedItemIndex(index) {
-    this.foundation_.setSelectedIndex(index);
+  /** @param {boolean} quickOpen */
+  set quickOpen(quickOpen) {
+    this.menuSurface_.quickOpen = quickOpen;
   }
 
-  /** @return {number} */
-  get selectedItemIndex() {
-    return this.foundation_.getSelectedIndex();
+  /** @param {boolean} isFixed */
+  set fixed(isFixed) {
+    this.menuSurface_.setFixedPosition(isFixed);
   }
 
-  /** @param {!boolean} rememberSelection */
-  set rememberSelection(rememberSelection) {
-    this.foundation_.setRememberSelection(rememberSelection);
+  hoistMenuToBody() {
+    this.menuSurface_.hoistMenuToBody();
   }
 
-  /** @param {boolean} quickOpen */
-  set quickOpen(quickOpen) {
-    this.foundation_.setQuickOpen(quickOpen);
+  /** @param {boolean} isHoisted */
+  setIsHoisted(isHoisted) {
+    this.menuSurface_.setIsHoisted(isHoisted);
+  }
+
+  /**
+   * Sets the element that the menu-surface is anchored to.
+   * @param {!HTMLElement} element
+   */
+  setAnchorElement(element) {
+    this.menuSurface_.anchorElement = element;
+  }
+
+  registerListeners_() {
+    this.root_.addEventListener('keydown', this.handleKeydown_);
+    this.root_.addEventListener('click', this.handleClick_);
+  }
+
+  deregisterListeners_() {
+    this.root_.removeEventListener('keydown', this.handleKeydown_);
+    this.root_.removeEventListener('click', this.handleClick_);
+  }
+
+  handleAfterOpened_() {
+    const list = this.items;
+    if (list.length > 0) {
+      list[0].focus();
+    }
   }
 
   /** @return {!MDCMenuFoundation} */
   getDefaultFoundation() {
     return new MDCMenuFoundation({
-      addClass: (className) => this.root_.classList.add(className),
-      removeClass: (className) => this.root_.classList.remove(className),
-      hasClass: (className) => this.root_.classList.contains(className),
-      hasNecessaryDom: () => Boolean(this.itemsContainer_),
-      getAttributeForEventTarget: (target, attributeName) => target.getAttribute(attributeName),
-      getInnerDimensions: () => {
-        const {itemsContainer_: itemsContainer} = this;
-        return {width: itemsContainer.offsetWidth, height: itemsContainer.offsetHeight};
+      addClassToElementAtIndex: (index, className) => {
+        const list = this.items;
+        if (list.length > index) {
```

Comment:
Questions about these various adapters that check `length > index`:

1. Should the adapters even need to sanity-check this? Should it even be possible for the foundation to pass an invalid index to these?
2. If the sanity check is needed, should we be checking both boundaries rather than just the high end?

Reply:
I'm probably being overly defensive here. I'll remove the check.